web_server/posts/views.py

```diff
@@ -141,13 +141,49 @@ def html_handler(request, posts, pager, pagination_uris):
     return endpoint.resolve()
 
 def comments_retrieval_and_creation_to_post_id(request, post_id):
+    def json_handler(request, posts, pager, pagination_uris):
+        # Explicitly add authors to the serialization
+        author_exclude_fields = ('password',"is_superuser", "is_staff", "groups", "user_permissions")
+
+        # Get the comments
+        comments = Comment.objects.filter(parentPost=posts[0])
+        comments_json = loads(serializers.serialize('json', comments))
+        comments_author_json = loads(serializers.serialize('json_e', [comment.author for comment in comments], exclude_fields=author_exclude_fields))
+        # Explicitly add authors to the comments
+        for j in range(0, len(comments_json)):
+            comments_json[j]['fields']['author'] = comments_author_json[j]['fields']  # avoid inserting meta data
+        # Strip meta data from each comment
+        for comment in comments_json:
+            comment['fields']['id'] = comment['pk']
+        comments_json = [comment['fields'] for comment in comments_json]
+
+        output = {
+            "query": "comments",
+            "count": pager.count,
+            "size": len(posts),
+            "comments": comments_json
+        }
+
+        (prev_uri, next_uri) = pagination_uris
+        if prev_uri:
+            output['prev'] = prev_uri
+        if next_uri:
+            output['next'] = next_uri
+
+        return JsonResponse(output)
+
     if request.method == 'POST':
         # JSON post body of what you post to a posts' comemnts
         # POST to http://service/posts/{POST_ID}/comments
         return HttpResponse("http://service/posts/{POST_ID}/comments POST")
     elif request.method == 'GET':
         # access to the comments in a post
```

Comment:
This logic is not required if you are using the endpoint handler to determine which handler will service the request.

Reply:
Changed the logic